### PR TITLE
Integrate Paired 2.0 Functionality: Notifications, Cancelations, and Statistics

### LINKED ISSUE #36: add ability for user to see past pairing history
- History component
- Able to see how many hours they have paired

### LINKED ISSUE #41: ability to cancel already booked pairings - as a pairee

### LINKED ISSUE #52: Email Notifications for Bookings/Account Sign Up
Instead of relying on having a page open, or allowing for Chrome notification permission on the Client's end, what if basic email notifications were implemented? This would also pull double duty of automatically implementing into people's GCal/iCal/Outlook when they receive it.
Sendgrid has a free tier: https://sendgrid.com/pricing/
But it looks like MailGun would be more scalable for these kinds of transactional emails: https://www.mailgun.com/

Additionally, it might be a good idea to include a, "Welcome!" email confirmation upon signing up as well.

### LINKED ISSUE #58: Notifications
Hey! I remember that you guys aren't quite ready to have it submit notifications but if there could be some kind of prompt for the person booking to then send a Slack message to whomever they're booking with, that would be great! I've had two people book with me and neither one has sent me a message. One followed up with me at 7:45 this morning  for an 8am meeting today! Needless to say, I couldn't show up with such short notice. Thanks in advance!

### PR DESCRIPTION
**Pull request by DanielEFrampton**
# Description
Please include a summary of the change and which issue is fixed:

- GraphQL queries are now sent to a [Rails back-end](https://github.com/DanielEFrampton/paired-be) which communicates to a [Flask micro-service](https://github.com/msimon42/paired-sms) and 3rd party APIs to send SMS notifications (or email if no phone number is provided):
    - to mentors when pairing sessions are booked with them;
    - to either the mentor or mentee when the other party cancels a pairing session;
    - to mentors with reminders of pairing sessions the morning of the day they occur.
- Users have the option to provide a phone number when creating an account or editing their profile.
- Pairing sessions can be canceled by either the mentor or mentee, returning that time slot to the mentor's open availability.
- Users can view statistics about their pairing history on a page accessible from the drop-down menu.
- The database system has transitioned from MongoDB to PostgreSQL, but the GraphQL schema and queries remain the same (with small exceptions due to syntax differences between Apollo Server and GraphQL-Ruby).
- Overhauled documentation with new exhaustive reference of GraphQL queries and example responses and up-to-date local deployment setup guide.

List any dependencies that are required for this change:

No new dependencies needed for new functionality, but the dependencies did get an overhaul to resolve issues with outdated packages which prevented successful builds and deployment.

Fixes or Closes [Issue #117 on paired-be](https://github.com/DanielEFrampton/paired-be/issues/117)
Resolves #58, resolves #52, resolves #41, resolves #36.

## Where should the reviewer start?:
In addition to the new Stats container, the majority of changes were to the Schedule container, Profile container, and the HamburgerMenu and Header components. See also the revised and new queries in `src/queries/index.js`.

## Additional information for the reviewer:

**Before merging**, we need to do a fresh import of data from MongoDB to our PostgreSQL database, and we would like to send out release notes to existing users notifying them about how the notifications work.

The root README file has been changed to that of what we've called the "Paired 2.0" project, but we've retained the original README, which is linked to from the root README, and credited the original dev team alongside our own. We are open to feedback about this and wish to represent both team's work in whatever way is best.

Also, we are primarily Rails developers and were learning React/Redux in the midst of this project. When we weren't throwing paint at the wall, the majority of our changes in this repo were accomplished by imitating existing patterns. If there are ways we could improve things, please let us know; we're open to feedback and wish to improve.

### CODE REVIEW

File: local_setup_guide.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Setting Up Paired Development Environment
+
+## Front-end Setup
+
+Clone down the [front-end repo](https://github.com/DanielEFrampton/paired-fe):
```
Comment (hillstew) [inline local_setup_guide.md]: Would this still be to your github after this merge to setup the development environment?

File: src/containers/Stats/index.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      this.props.setError(error.message);
+    }
+  }
+
+  render() {
+    const stats = this.state.stats;
+    return (
+      <div className='Stats'>
+        <h2 className='Stats--h2'>Stats for {stats.name}   <span role='img' aria-label='upward chart emoji'>
+        📈 </span></h2>
```
Comment (hillstew) [inline src/containers/Stats/index.js]: love the emoji here!

File: src/containers/Stats/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  componentDidMount() {
+    this.getUserStats();
+  }
+
+  getUserStats = async () => {
+    const id = this.props.user.id;
+    const statsQuery = gql.getUserStats(id);
+    try {
+      const statsResponse = await fetchData(statsQuery);
```
Comment (hillstew) [inline src/containers/Stats/index.js]: nice reuse of the existing fetchData!

File: local_setup_guide.md (hunk 1 above)
Comment (DanielEFrampton) [reply local_setup_guide.md]: Whoops nope. Fix incoming.

File: src/containers/Stats/index.js (hunk 3 above)
Comment (DanielEFrampton) [reply src/containers/Stats/index.js]: Thanks!

**Pull request merged**